### ISSUE TITLE
Easy way to publish workspaces/wizards

### ISSUE THREAD

**Issue by darabos** [lynxkite/lynxkite#148]
Currently you have to copy the boxes into a YAML file and the recipient has to paste it into a LynxKite instance and then fix the imports. (God be with you if you have custom boxes.) Or you have to run a public LynxKite instance. Neither is very friendly!

Ideally you want to be able to build something, publish it, and send a link to someone. They can click the link and see your work.

What could we do?
- **Workspace bundles.** This is already on the roadmap. We would wrap up the data and all workspaces and you can drop it into another LynxKite instance. Not everyone has a LynxKite instance. They could use https://demo.lynxkite.com/, but it's pretty far from the ideal.
- **A shared public instance.** Keeping people from accidentally destroying this would be a challenge. We would also be liable for policing it.
- **Publish as HTML.** Maybe we could take a workspace/wizard and record all backend responses and embed it all in a file. Then you can just host that file wherever.
- **Publish as YAML with a viewer.** Same us the last one, but the file would not include the whole LynxKite UI. Instead we would run a service like https://nbviewer.jupyter.org/ to display the files when they are hosted as a GitHub Gist or something. You would link to https://viewer.lynxkite.com/gist.github.com/darabos/xxxx.

I like this last option, as it would allow us to apply UI updates to works published in the past. It could also work as part of a community hub. Yet it wouldn't cost compute or hosting on our side.

**Comment by xandrew-lynx** [comment]
Yeah, I agree this is an important problem and we are not doing great right now.

Two questions on the solution options.

First, do I get it right that the last two options would only work for practically static wizards? If there is significant interactivity then these seem hopeless, right? But for the non-interactive case, I see little benefit of wizards relative to, say, a ppt, so I'm not sure I'd invest much in a solution which works only for static wizards.

Second, you say the last option wouldn't cost compute on our side. How come? Wouldn't we need to host a viewer service?

**Comment by darabos** [comment]
> Yeah, I agree this is an important problem and we are not doing great right now.
> 
> Two questions on the solution options.
> 
> First, do I get it right that the last two options would only work for practically static wizards? If there is significant interactivity then these seem hopeless, right? But for the non-interactive case, I see little benefit of wizards relative to, say, a ppt, so I'm not sure I'd invest much in a solution which works only for static wizards.

Correct, it's only for static wizards. It's similar to Jupyter Notebooks. There is absolutely non-zero value in being able to look at them.

> Second, you say the last option wouldn't cost compute on our side. How come? Wouldn't we need to host a viewer service?

But that's basically a static page. It can probably fetch the YAML in the browser and doesn't need any backend. Importantly, it doesn't need to execute any LynxKite operations!
